File: src/main/scala/com/bombproj/framework/SpringContextHolder.java
package com.bombproj.framework;

import org.springframework.beans.BeansException;
import org.springframework.context.ApplicationContext;
import org.springframework.context.ApplicationContextAware;
import org.springframework.stereotype.Component;

@Component
public class SpringContextHolder implements ApplicationContextAware {

    private static ApplicationContext context;

    @Override
    public void setApplicationContext(ApplicationContext applicationContext) throws BeansException {
        SpringContextHolder.context = applicationContext;
    }

    public static <T> T getBean(Class<T> beanClass) {
        return context.getBean(beanClass);
    }

}

File: src/main/scala/com/bombproj/framework/SpringContextHolder.scala
package com.bombproj.framework

import com.bombproj.framework.SpringContextHolder.context
import org.springframework.context.{ApplicationContext, ApplicationContextAware}
import org.springframework.stereotype.Component

@Component
class SpringContextHolder extends ApplicationContextAware {

    override def setApplicationContext(applicationContext: ApplicationContext): Unit = {
        context = applicationContext
    }

}

object SpringContextHolder {
    private var context: ApplicationContext = _

    def getBean[T](clazz: Class[T]): T = {
        context.getBean(clazz)
    }
}
